preprocessor/pipeline.json

```diff
@@ -0,0 +1,34 @@
+{
+    "pipeline": {
+        "dataset": {
+            "label": "mnist",
+            "type": "image"
+        },
+        "operations": [
+            {
+                "map": [
+                    {"map_func": "normalize_img"},
+                    {"num_parallel_calls": "autotune"}
+                ]
```

Comment:
Is there a reason that the arguments in the function are lists with JSON objects inside? Do they need to be ordered? If not, this can be made less verbose by making the following change:
```
"map": [
                    {"map_func": "normalize_img"},
                    {"num_parallel_calls": "autotune"}
                ]
```
to
```
"map": {
                    "map_func": "normalize_img",
                    "num_parallel_calls": "autotune"
             }
```

This reduces the nesting in the JSON

Reply:
Yea they do. If you look at `map_variables()` I need to be able to iterate over the variables and replace their value string with what the value actually should be.

Reply:
Well you can still iterate over a dictionary, it just doesn't guarantee order of the arguments. The other thing I noticed was that each `op` object has a different JSON structure. By that I mean the key for each value is different, requiring you to convert to a list to get the operation. You can eliminate the casting by make only the values in the JSON dynamic

Reply:
I'm going to add suggestions to work with that structure